src/v/kafka/server/write_at_offset_stm.cc

```diff
@@ -151,24 +162,35 @@ ss::future<result<raft::replicate_result>> write_at_offset_stm::do_replicate(
         co_return make_error_code(errc::invalid_offset);
     }
 
-    chunked_vector<model::record_batch> batches;
-    // fill the gap with ghost batches if required
-    if (effective_prev_log_offset != kafka::prev_offset(expected_base_offset)) {
-        // the term argument is not important here, it will be
-        // overriden in Raft layer
-        auto ghost_batches = model::make_ghost_batches(
-          kafka::offset_cast(kafka::next_offset(stm_last_offset)),
-          model::prev_offset(kafka::offset_cast(expected_base_offset)),
-          model::term_id{0});
-        batches.reserve(ghost_batches.size() + 1);
-        std::ranges::move(ghost_batches, std::back_inserter(batches));
+    chunked_vector<model::record_batch> effective_batches;
+    auto effective_last_offset = effective_prev_log_offset;
+    // Fill in any gaps in the input batch offset space.
+    for (size_t i = 0; i < expected_base_offsets.size(); ++i) {
+        auto expected_prev_offset = kafka::prev_offset(
+          expected_base_offsets[i]);
+        if (effective_last_offset != expected_prev_offset) {
+            // fill with ghost batch
```

Comment:
nitpick: I think the "term argument is not important..." comment was a useful one

Reply:
yep, will restore.